src/rules/selector-pseudo-element-no-unknown/index.js

```diff
@@ -0,0 +1,67 @@
+import { isString } from "lodash"
+import selectorParser from "postcss-selector-parser"
+import { vendor } from "postcss"
+import {
+  isKnownPseudoElement,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "selector-pseudo-element-no-unknown"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (u) => `Unexpected unknown pseudo-element selector "${u}"`,
+})
+
+// level 1 and 2 pseudo elements
+const pseudoElements = new Set([
+  ":before", ":after", ":first-line", ":first-letter",
+])
+
+export default function (actual, options) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, { actual }, {
+      actual: options,
+      possible: {
+        ignorePseudoElements: [isString],
+      },
+      optional: true,
+    })
+    if (!validOptions) { return }
+
+    root.walkRules(rule => {
+      const selector = rule.selector
+
+      if (selector.indexOf(":") === -1) { return }
+
+      selectorParser(selectorTree => {
+        selectorTree.eachPseudo(pseudoNode => {
+          const pseudoElement = pseudoNode.value
+          const pseudoElementLowerCase = pseudoElement.toLowerCase()
+
+          if (vendor.prefix(pseudoElementLowerCase.replace("::", ""))) { return }
+
+          if (pseudoElement.indexOf("::") === -1
+            && !pseudoElements.has(pseudoElementLowerCase)) {
+            return
+          }
+
+          if (isKnownPseudoElement(pseudoElementLowerCase)) { return }
```

Comment:
So will the need for these two separate lists go away if the rule only supports two colons?